src/Compilers/Core/Portable/CommandLine/CommonCompiler.cs

```diff
@@ -132,7 +132,32 @@ internal virtual string GetAssemblyFileVersion()
             {
                 var name = $"{Type.GetTypeInfo().Assembly.GetName().Name}.dll";
                 var filePath = Path.Combine(_clientDirectory, name);
-                return FileVersionInfo.GetVersionInfo(filePath).FileVersion;
+                var fileVersionInfo = FileVersionInfo.GetVersionInfo(filePath);
+
+                return $"{fileVersionInfo.FileVersion} ({ExtractShortSHA(fileVersionInfo.ProductVersion)})";
+            }
+
+            return "";
+        }
+
+        internal static string ExtractShortSHA(string productVersion)
+        {
+            if (productVersion != null)
+            {
+                string marker = "Commit Hash: "; // format used from AssemblyVersionAttribute
```

Comment:
Can you be more specific? 
I looked at attributes for adding information to an assembly and didn't find spare ones. Also, as you can see from the screenshot, all FileVersionInfo are used.